src/admin/components/forms/FieldDescription/index.tsx

```diff
@@ -11,13 +11,14 @@ const FieldDescription: React.FC<Props> = (props) => {
     className,
     description,
     value,
+    formatOptions,
   } = props;
 
   const { i18n } = useTranslation();
 
   if (isComponent(description)) {
     const Description = description;
-    return <Description value={value} />;
+    return <Description value={formatOptions ? new Intl.NumberFormat(i18n.language, formatOptions).format(value as number) : value} />;
```

Comment:
This doesn't seem right. We should not be formatting the description at all, especially considering it's accepts a callback so _you_ can format it. This also doesn't protect field types that are not numbers from running through `Intl.NumberFormat`.

Reply:
I disagree here @jacobsfletch, this is not formatting the description, only the value passed to it. And at the moment, the description functions or components aren't being passed the `i18n.language`. If the formatted values aren't passed to the Description, the language would need to be passed to it at the very least.

And to your second comment, only number fields have `formatOptions` at the moment.

Reply:
Passing the language through the description callback sounds fine to me. Then back to that second point, you are typing the value `as number` which clearly illustrates the problem.

Reply:
The latest commits have pulled formatting out of the Description, and added the `language` argument to the description function.